golang/gorm/src/controller/orderController.go

```diff
@@ -0,0 +1,144 @@
+package controller
+
+import (
+	"model"
+	"github.com/gorilla/mux"
+	"github.com/satori/go.uuid"
+	"fmt"
+	"encoding/json"
+	"net/http"
+)
+
+/* handler function for GET method */
+var SelectOrders = func (w http.ResponseWriter, r *http.Request) {
+	orders := []model.Order{}
+	err := GetDB().Table("orders").Find(&orders).Error
+	if err != nil {
+		msg := map[string]interface{} {"status" : false, "message" : err}
+		w.Header().Add("Content-Type", "application/json")
+		json.NewEncoder(w).Encode(msg)
+		return
+	}
+	
+	w.Header().Add("Content-Type", "application/json")
+	json.NewEncoder(w).Encode(orders)
+}
+
+/* handler function for GET method */
+var SelectOrder = func (w http.ResponseWriter, r *http.Request) {
+	orderId := mux.Vars(r)["orderId"]
+	fmt.Println(fmt.Sprintf("finding order with order_id = %s", orderId))
```

Comment:
Why is fmt.Printf() not preferable to fmt.Println(fmt.Sprintf()) ? Is this in anticipation of using the intermediate string for some other purpose as well ?